### PR TITLE
`asn1/x_algor.c`: Add internal `ossl_X509_ALGOR_from_nid()`

### PR DESCRIPTION
**Pull request by DDvO**
This simplifies internal code

### CODE REVIEW

File: apps/cmp.c [hunk 1]
```diff
@@ -2803,11 +2811,13 @@ int cmp_main(int argc, char **argv)
...
-         || opt_tls_keypass != NULL || opt_tls_extra != NULL
-         || opt_tls_trusted != NULL || opt_tls_host != NULL)
-            && !opt_tls_used)
-        CMP_warn("Ingnoring TLS options(s) since -tls_used is not given");
+    if (opt_tls_cert == NULL && opt_tls_key == NULL && opt_tls_keypass == NULL
+            && opt_tls_extra == NULL && opt_tls_trusted == NULL
+            && opt_tls_host == NULL) {
+        if (opt_tls_used)
+            CMP_warn("-tls_used given without any other TLS options");
+    } else if (!opt_tls_used)
```
Comment (paulidale) [inline apps/cmp.c:2819]: style nit: {} around this block too.

**Review by paulidale — APPROVED** [review]
LGTM after a stlye nit is addressed.

Is there meant to be the extra clean up included as well?

**Comment by t8m** [comment]
The commit 15e2b0f (or rather it's equivalent commit ad1a1d71) is present in master already. @DDvO can you please rebase this PR?

File: apps/cmp.c (hunk 1 above)
Comment (DDvO) [reply apps/cmp.c:2819]: Fixed, as well as a couple further coding style nits reported by `check-format.pl` (I should have used this tool again before),
in a separate commit.
Wanna re-approve?

**Comment by DDvO** [comment]
> The commit [15e2b0f](https://github.com/openssl/openssl/commit/15e2b0f02d5e4fcc683e9e14539ab5b8187ca754) (or rather it's equivalent commit [ad1a1d7](https://github.com/openssl/openssl/commit/ad1a1d715dcab875dafd6e792b8eb65eb84d6b9f)) is present in master already. @DDvO can you please rebase this PR?

Sorry, also this must have been due to may recent confusion of branches; fixed by rebasing.

**Comment by DDvO** [comment]
Is this ok to merge now,
or does anyone want to review and (re-)approve the style nit fixes I added to `apps/cmp.c`?

**Comment by t8m** [comment]
This needs to be re-approved.

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @t8m and @paulidale

**Pull request closed without merging**